### PR TITLE
chore: reduce cloud-init var overhead for VHD scenarios

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR reduces the ARM template overhead in non-ubuntu distro (i.e., VHD) scenarios. The vars that we deliver to populate cloud-init are only needed when cloud-init will actually consume them.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -1045,7 +1045,10 @@ func (p *Properties) IsUbuntuDistroForAllNodes() bool {
 			}
 		}
 	}
-	return p.MasterProfile.Distro == Ubuntu || p.MasterProfile.Distro == Ubuntu1804
+	if p.MasterProfile != nil {
```
Comment (CecileRobertMichon) [inline pkg/api/types.go]: Good catch. Do we have a unit test case that covers the nil p.MasterProfile scenario?

File: pkg/engine/armvariables.go [hunk 2]
```diff
@@ -147,6 +147,60 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		"vnetResourceGroupNameResourceSegmentIndex": 4,
 	}
 
+	if cs.Properties.IsUbuntuDistroForAllNodes() {
+		masterVars["cloudInitFiles"] = map[string]interface{}{
```
Comment (CecileRobertMichon) [inline pkg/engine/armvariables.go]: slight preference for doing something like:

```
masterVars["cloudInitFiles"] = map[string]interface{}{
			"provisionScript":                  getBase64EncodedGzippedCustomScript(kubernetesCSEMainScript),
			"provisionSource":                  getBase64EncodedGzippedCustomScript(kubernetesCSEHelpersScript),
			"provisionInstalls":                getBase64EncodedGzippedCustomScript(kubernetesCSEInstall),
			"provisionConfigs":                 getBase64EncodedGzippedCustomScript(kubernetesCSEConfig),
			"provisionCIS":                     getBase64EncodedGzippedCustomScript(kubernetesCISScript),
			"healthMonitorScript":              getBase64EncodedGzippedCustomScript(kubernetesHealthMonitorScript),
			"customSearchDomainsScript":        getBase64EncodedGzippedCustomScript(kubernetesCustomSearchDomainsScript),
			"generateProxyCertsScript":         getBase64EncodedGzippedCustomScript(kubernetesMasterGenerateProxyCertsScript),
			"mountEtcdScript":                  getBase64EncodedGzippedCustomScript(kubernetesMountEtcd),
			"kubeletSystemdService":            getBase64EncodedGzippedCustomScript(kubeletSystemdService),
			"kmsSystemdService":                getBase64EncodedGzippedCustomScript(kmsSystemdService),
			"kubeletMonitorSystemdTimer":       getBase64EncodedGzippedCustomScript(kubernetesKubeletMonitorSystemdTimer),
			"kubeletMonitorSystemdService":     getBase64EncodedGzippedCustomScript(kubernetesKubeletMonitorSystemdService),
			"dockerMonitorSystemdTimer":        getBase64EncodedGzippedCustomScript(kubernetesDockerMonitorSystemdTimer),
			"dockerMonitorSystemdService":      getBase64EncodedGzippedCustomScript(kubernetesDockerMonitorSystemdService),
			"aptPreferences":                   getBase64EncodedGzippedCustomScript(aptPreferences),
			"dockerClearMountPropagationFlags": getBase64EncodedGzippedCustomScript(dockerClearMountPropagationFlags),
			"etcdSystemdService":               getBase64EncodedGzippedCustomScript(etcdSystemdService),
		}

if cs.Properties.IsUbuntuDistroForAllNodes() {
     // append all the ubuntu distro specific vars
}

```

That way there are no duplicated lines and we avoid risking someone making a mistake and only adding a new var for one of the two cases.

Comment (jackfrancis) [reply pkg/engine/armvariables.go]: image

**Comment by jackfrancis** [comment]
Validated that this works w/ "ubuntu" distro

File: parts/k8s/cloud-init/masternodecustomdata.yml [hunk 3]
```diff
@@ -112,21 +112,21 @@ write_files:
     {{CloudInitData "aptPreferences"}}
 
 {{if .MasterProfile.IsUbuntuNonVHD}}
-{{if .MasterProfile.IsUbuntu1604}}
+  {{if .MasterProfile.IsUbuntu1604}}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/masternodecustomdata.yml]: Added some indentation here for readability

File: pkg/engine/armvariables.go (hunk 2 above)
Comment (CecileRobertMichon) [reply pkg/engine/armvariables.go]: `masterVars["cloudInitFiles"].(map[string]interface{})["etcIssue"] = getBase64EncodedGzippedCustomScript(etcIssue)` should work

File: pkg/api/types_test.go [hunk 4]
```diff
@@ -1252,6 +1252,28 @@ func TestHasUbuntuDistroForAllNodes(t *testing.T) {
 			},
 			expected: false,
 		},
+		{
```
Comment (jackfrancis) [inline pkg/api/types_test.go]: added nil master unit test scenarios for `IsUbuntuDistroForAllNodes` func

**Comment by jackfrancis** [comment]
@CecileRobertMichon this is ready for final review

**Pull request merged**